docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+cd android
+./gradlew generateCodegenArtifactsFromSchema --rerun-tasks
+```
+
+This script first adds the package to the app, in the same way iOS does. Then, after moving to the `android` folder, it invokes a Gradle task to create the generated code.
+
+:::note
+To run **CodeGen**, you need to enable the **New Architecture** in the Android app. This can be done by opening the `gradle.properties` files and by switching the `newArchEnabled` property from `false` to `true`.
+:::
+
+The generated code is stored in the `MyApp/node_modules/rtn-calculator/android/build/generated/source/codegen` folder and it has this structure:
+
+TODO: ordering issue. I can't actually get codegen to work at this point. Only works once I add the native files following. Probably similar issue for Fabric guide. Including generated file structure for later ease of use:
+
+```title="Android generated code"
+codegen
+├── java
+│   └── com
+│       └── RTNCalculator
+│           └── NativeCalculatorSpec.java
+├── jni
+│   ├── Android.mk
+│   ├── RTNCalculator-generated.cpp
+│   ├── RTNCalculator.h
+│   └── react
+│       └── renderer
+│           └── components
+│               └── RTNCalculator
+│                   ├── ComponentDescriptors.h
+│                   ├── EventEmitters.cpp
+│                   ├── EventEmitters.h
+│                   ├── Props.cpp
+│                   ├── Props.h
+│                   ├── ShadowNodes.cpp
+│                   └── ShadowNodes.h
+└── schema.json
+```
+
+#### Write the Native Android Code
+
+The native code for the Android side of a TurboModule requires three more files:
+
+1. An `AndroidManifest.xml` file.
+2. A `RTNCalculatorModule.java` that implements the module.
+4. A `RTNCalculatorPackage.java` that React Native uses to configure the library.
+
+The final structure within the Android library should look like this:
+
+```title="Android Folder Structure"
+android
+├── build.gradle
+└── src
+    └── main
+        ├── AndroidManifest.xml
+        └── java
+            └── com
+                └── RTNCalculator
+                    ├── CalculatorModule.java
+                    └── CalculatorPackage.java
+```
+
+##### AndroidManifest.xml
+
+```xml title="AndroidManifest.xml"
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="com.RTNCalculator">
+</manifest>
+```
+
+This is a small manifest file that defines the package for our module.
+
+
+##### CalculatorModule.java
+
+```java title="CalculatorModule.java"
+package com.RTNCalculator;
+
+import androidx.annotation.NonNull;
+import com.facebook.react.bridge.NativeModule;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import java.util.Map;
+import java.util.HashMap;
+
+public class CalculatorModule extends NativeCalculatorSpec {
+
```

Comment:
Let's add a:
```java
public static String NAME = "RTNCalculator";
```

and replace all the strings both here and inside `CalculatorPackage` with this static field.